### PR TITLE
metrics: add infoschema v2 cache hit/miss/evict

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref #50959

Problem Summary:

### What changed and how does it work?

Add metrics for infoschema v2 cache hit/miss/evict.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
  image
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: pkg/metrics/grafana/tidb.json [hunk 1]
```diff
@@ -13902,6 +13902,112 @@
...
+          "gridPos": {
+            "x": 12,
+            "y": 31,
+            "w": 12,
+            "h": 7
+          },
+          "id": 23763572012,
+          "targets": [
+            {
+              "expr": "sum(rate(tidb_domain_infoschema_v2_cache{k8s_cluster=\"$k8s_cluster\", tidb_cluster=\"$tidb_cluster\", instance=~\"$instance\"}[1m])) by (type)",
```
Comment (D3Hunter) [inline pkg/metrics/grafana/tidb.json:13918]: wound be better to calculate a hit/miss/evict percent

**Review by D3Hunter — APPROVED** [review]

File: pkg/metrics/grafana/tidb.json (hunk 1 above)
Comment (tangenta) [reply pkg/metrics/grafana/tidb.json:13955]: https://github.com/pingcap/tidb/pull/54268/commits/11b48a420c8ddc5dc91024149c95e9981a5c4e26

Comment (tangenta) [reply pkg/metrics/grafana/tidb.json:13955]: image

**Pull request merged**